app/models/notice.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class Notice < ActiveModelSerializers::Model
+  attributes :id, :icon, :title, :message, :actions
+
+  # Notices a user has seen are stored as a bitmap in
+  # `users.seen_notifications`.
+  NOTICE_BIT_MAP = {
+    mastodon_privacy_4_2: 1, # rubocop:disable Naming/VariableNumber
+  }.freeze
+
+  def dismiss_for_user!(user)
+    user.update!(seen_notices: (user.seen_notices || 0) | NOTICE_BIT_MAP[id])
+  end
+
+  class Action < ActiveModelSerializers::Model
+    attributes :label, :url
+  end
+
+  class << self
+    include RoutingHelper
+
+    def first_unseen(user)
+      notice_key = NOTICE_BIT_MAP.find { |_, bit| ((user.seen_notices || 0) & bit).zero? }&.first
+
+      send("#{notice_key}_notice") if notice_key.present?
+    end
+
+    def find(key)
+      throw ActiveRecord::RecordNotFound unless NOTICE_BIT_MAP.key?(key.to_sym)
+
+      send("#{key}_notice")
+    end
+
+    private
+
+    def mastodon_privacy_4_2_notice
```

Comment:
This part of the code looks a little odd, seems like it shouldn't really need the "new" style of constructor based on `send` — just an hash of notice to actions should be enough?

Reply:
I decided to do it this way so that values are translated, and decided against doing something more declarative so that `i18n-tasks` can properly pick up which translation strings are used, but I guess there could be better ways?